### PR TITLE
2016-2017 Lesson Overhaul

### PR DESCRIPTION
**Pull request by ElijahCaine**
**NOTE:** I know it's a big diff, a lot of the content previously in the repository will be added back in (images, etc).

## Major Changes Include:
- Organized all content to focus on lessons.
  - In the side-bar you can now see Lessons in order.
  - Clicking the 'next' button goes to the next lesson.
- Re-writing all content.
  - Re-writing content with a focus on HTML pages over of slides.
- De-constructed lessons to be smaller.
  - This is so we can pick, choose, and move lessons around more easily.

## TODO:
- [ ] index
  - [X] Drafted
  - [x] Edited
- [ ] start-here
  - [X] Drafted
  - [x] Edited
  - [x] HTML & Slides Look good
- [ ] first-steps
  - [X] Drafted
  - [x] Edited
  - [x] HTML & Slides Look good
- [ ] operating-systems
  - [X] Drafted
  - [x] Edited
  - [x] HTML & Slides Look good
- [ ] shell-navigation-os
  - [x] Drafted
  - [x] Edited
  - [x] HTML & Slides Look good
- [ ] users-groups-permissions
  - [x] Drafted
  - [x] Edited
  - [x] HTML & Slides Look good
- [ ] files
  - [x] Drafted
  - [x] Edited
  - [x] HTML & Slides Look good
- [ ] packages-software-libraries
  - [x] Drafted
  - [x] Edited
  - [x] HTML & Slides Look good
- [ ] questions-answers-documentation
  - [x] Drafted
  - [x] Edited
  - [x] HTML & Slides Look good
- [ ] version-control
  - [x] Drafted
  - [x] Edited
  - [x] HTML & Slides Look good
- [ ] programming
  - [x] Drafted
  - [x] Edited
  - [x] HTML & Slides Look good
- [ ] frameworks
  - [x] Drafted
  - [x] Edited
  - [x] HTML & Slides Look good
- [ ] testing
  - [x] Drafted
  - [x] Edited
  - [x] HTML & Slides Look good
- [ ] continuous-integration
  - [x] Drafted
  - [x] Edited
  - [x] HTML & Slides Look good
- [ ] security
  - [x] Drafted
  - [x] Edited
  - [x] HTML & Slides Look good
- [ ] databases
  - [x] Drafted
  - [x] Edited
  - [x] HTML & Slides Look good
- [ ] development-processes-tools
  - [x] Drafted
  - [x] Edited
  - [x] HTML & Slides Look good
- [ ] dns
  - [x] Drafted
  - [x] Edited
  - [x] HTML & Slides Look good
- [ ] configuration-management
  - [x] Drafted
  - [x] Edited
  - [x] HTML & Slides Look good
- [ ] application-isolation
  - [x] Drafted
  - [x] Edited
  - [x] HTML & Slides Look good
- [ ] cloud-infrastructure
  - [x] Drafted
  - [x] Edited
  - [x] HTML & Slides Look good
- [ ] contributing-to-open-source
  - [x] Drafted
  - [x] Edited
  - [x] HTML & Slides Look good
- [ ] about
  - [x] Drafted
  - [ ] Edited
- [ ] schedule
  - [ ] Drafted
  - [ ] Edited

## Contributing:

I have a few notes for people that want to help contribute:

### RST Formatting:

    This is the top Heading  (h1)
    =======================

    This is one topic  (h2)
    -----------------

    This is a subtopic  (h3)
    ~~~~~~~~~~~~~~~~~~

    This is another topic  (h2)
    ---------------------

    etc  (h3)
    ~~~

- Leave two newlines above *any* heading (lesson, topic, subtopic, sub-sub-topic).
- Use the `.. ifnotslides::` directive to include paragraphs for the HTML pages but not the slides.
- Lines have an 80 character limit.

### Adding Content

**If you want to fill out the lessons do the following**:
- Call dibs in this pull request.
- Reference the old content when filling in the new content. **DO NOT COPY-PASTE CONTENT**
- Feel free to expand on topics/subtopics where you feel it appropriate.

Follow git-flow. Before making changes `git pull` and new change and avoid conflicts. If we start stepping on eachother's toes too much we'll start making proper pull-requests to this branch.

Of course, feel free to edit where it is needed.

refs #120

### CODE REVIEW

**Comment by ElijahCaine** [comment]
I am going to do the drafts for:
- shell-navigation-os
- users-groups-permissions
- files

I am also adding an 'overview' section to each document.

**Comment by ElijahCaine** [comment]
Now I'm calling dibs on:
- packages-software-libraries
- questions-answers-documentation
- version control

**Comment by ElijahCaine** [comment]
I'm calling dibs on Programming, Frameworks, Testing, and CI.

**Comment by ElijahCaine** [comment]
Calling dibs on
- Security
- Databases
- Development Tools
- DNS

File: source/conf.py [hunk 1]
```diff
@@ -41,7 +41,7 @@
 
 # General information about the project.
 project = u'OSU DevOps BootCamp'
-copyright = u'2013, OSU OSL & OSU LUG'
+copyright = u'2013-2016, OSU Open Soruce Lab & OSU Linux Users Group'
```
Comment (tschuy) [inline source/conf.py]: `Soruce`

**Comment by ElijahCaine** [comment]
Calling dibs on

- configuration-management
- application-isolation
- cloud-infrastructure
- contributing-to-open-source

**Comment by aaroncohen73** [comment]
A bit of feedback for Lesson 13: Security:

When you suggest that people consider using pass phrases instead of passwords, you should probably also mention some of the pitfalls of pass phrases that might make them as vulnerable as passwords. For example, pass phrases should be made of random words, and *not* popular quotes, actor names, etc. It's not infeasible that a dictionary attack could target common phrases such as these.

**Pull request merged**